stencil/stencil.go

```diff
@@ -0,0 +1,64 @@
+package stencil
+
+import (
+	"fmt"
+	"path"
+	"sync"
+)
+
+type Stencil struct {
+	Reference string
+	valid     *bool
+	bundle    bundle
+	registry  *registry
+	runc      *runc
+}
+
+func NewStencil(ref string, registry *registry, runc *runc) *Stencil {
+	var once sync.Once
+	return &Stencil{
+		Reference: ref,
+		valid:     nil,
+		bundle: bundle{
+			path:  "",
+			once:  &once,
+			ready: make(chan interface{}),
+		},
+		registry: registry,
+		runc:     runc,
+	}
+}
+
+type bundle struct {
+	path  string
+	once  *sync.Once
+	ready chan interface{}
+}
+
+func (s *Stencil) loadBundle() {
+	go s.bundle.once.Do(func() {
+		bundle, err := s.registry.loadStencilBundle(s.Reference)
+		if err != nil {
+			panic(fmt.Errorf("failed to create stencil bundle :%s", err))
+		}
+		s.bundle.path = bundle
+		close(s.bundle.ready)
+	})
+}
```

Comment:
This is a neat apparatus, one that definitely benefits from more documentation and explanation of what is going on.  Specifically, the interplay between the sync.Once, the channel blocker, and the non-blocking uses of loadBundle().